### DISCUSSION TITLE
Control NODE_OPTIONS on deploy

### DISCUSSION THREAD

**Discussion by bizob2828** [vercel/community#3982] [Feature Question]
### Question

Hello, I work at New Relic on the Node.js agent team and we have instrumentation for server-side Next.js.  We have customers wanting to deploy their Next.js app on [Vercel](https://github.com/newrelic/newrelic-node-nextjs/issues/94). To load our instrumentation you have to provide `NODE_OPTIONS='-r @newrelic/next'` at runtime.  It does not appear that you can customize the run/deploy command.  I tried just setting `NODE_OPTIONS='-r @newrelic/next'` in the env vars but that runs at every phase and since the customer has yet to install deps that fails in build.  I then tried to call a local js file and get clever and inspect the process.argv to decide if you should load the `@newrelic/next` package but it says it cannot find the script.

We just need advice here on how to control either the run/deploy command or how to inject NODE_OPTIONS only on run/deploy.

**Comment by bizob2828** [comment]
Ok, I think I was approaching this from the wrong angle. It appears this solves my needs https://nextjs.org/docs/pages/building-your-application/optimizing/instrumentation

**Reply by bizob2828** [reply]
After further digging I don't think this will work due to how a Next.js app is deployed to vercel as different Vercel components.  You can see [here](https://nextjs.org/docs/pages/building-your-application/deploying), for the things our package instruments API routes are run as serverless functions, middleware is run as an edge function. I doubt the internals we use to monkey patch next server are in play here.

**Reply by stephenhmarsh** [reply]
@bizob2828 It might be worth looking at how Sentry achieved their profiler with: https://github.com/getsentry/profiling-node which works for us, on Vercel-hosted Next.js.

**Reply by amyegan** [reply]
I reached out to the integration engineers and they recommended OpenTelemetry: https://nextjs.org/docs/pages/building-your-application/optimizing/open-telemetry

The recommendation to look into Sentry's solution is also a nice idea. Thanks for sharing that, @stephenhmarsh!
